terraform/app.tf

```diff
@@ -0,0 +1,138 @@
+resource "azurerm_service_plan" "service-plan" {
+  name                = "getanid-authserver-${var.environment_name}-app-spln"
+  location            = data.azurerm_resource_group.group.location
+  resource_group_name = data.azurerm_resource_group.group.name
+  os_type             = "Linux"
+  sku_name            = var.app_service_plan_sku
+
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+resource "azurerm_linux_web_app" "auth-server-app" {
+  name                = local.get_an_identity_app_name
+  location            = data.azurerm_resource_group.group.location
+  resource_group_name = data.azurerm_resource_group.group.name
+  service_plan_id     = azurerm_service_plan.service-plan.id
+  https_only          = true
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  site_config {
+    http2_enabled       = true
+    minimum_tls_version = "1.2"
+  }
+
+  app_settings = {
+    HOSTING_ENVIRONMENT                    = var.environment_name,
+    APPLICATION_INSIGHTS_CONNECTION_STRING = azurerm_application_insights.insights.connection_string
+    POSTGRES_CONNECTION_STRING             = "Host=${local.postgres_server_name};Username=${local.infrastructure_secrets.POSTGRES_ADMIN_USERNAME};Password=${local.infrastructure_secrets.POSTGRES_ADMIN_PASSWORD};Database=${local.postgres_database_name}"
+    REDIS_URL                              = "${azurerm_redis_cache.redis.hostname}/${azurerm_redis_cache.redis.primary_access_key}",
+    AZURE_STORAGE_ACCOUNT_NAME             = azurerm_storage_account.data-protection.name,
+    AZURE_STORAGE_ACCESS_KEY               = azurerm_storage_account.data-protection.primary_access_key,
+    AZURE_STORAGE_CONTAINER                = azurerm_storage_container.uploads.name
+  }
+
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+resource "azurerm_linux_web_app" "test-server-app" {
+  count               = var.environment_name == "dev" ? 1 : 0
```

Comment:
Could we create a variable in de.tfvars.json instead and call it somthing like "deploy_test_server_app"? this would be true for dev and default to false